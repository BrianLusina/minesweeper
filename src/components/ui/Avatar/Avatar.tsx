import { FunctionComponent } from 'react';
import logo from '@images/logo.png';
import { AvatarProps } from './Avatar.interface';

const Avatar: FunctionComponent<AvatarProps> = ({ url }) => {
  return (
    <span className="image avatar">
      <img src={url || logo} alt="avatar" />
    </span>
  );
};

export default Avatar;
