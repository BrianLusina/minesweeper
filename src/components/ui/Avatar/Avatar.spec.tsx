import { render } from '@testing-library/react';
import Avatar from './Avatar';

describe('Avatar', () => {
  it('should render', () => {
    const url = 'https://avatar.png';

    const { container } = render(<Avatar url={url} />);

    expect(container).toMatchSnapshot();
  });
});
