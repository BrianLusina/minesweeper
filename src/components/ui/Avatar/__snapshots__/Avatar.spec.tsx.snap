// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render 1`] = `
<div>
  <span
    class="image avatar"
  >
    <img
      alt="avatar"
      src="https://avatar.png"
    />
  </span>
</div>
`;
